import "type.proto";
package pbnavitia;

message PlacesRequest {
    required string q = 1;
    repeated NavitiaType types = 2;
    required int32 depth = 3;
    required int32 nbmax = 4;
    repeated string admin_uris = 5;
}

message NextStopTimeRequest {
    required string departure_filter = 1;
    required string arrival_filter = 2;
    required string from_datetime = 3;
    required int32 duration = 4;
    required int32 depth = 5;
    required int32 nb_stoptimes = 7;
}

message StreetNetworkParams{
    optional string origin_mode = 1;
    optional string destination_mode = 2;
    optional double walking_speed = 3;
    optional int32 walking_distance =4;
    optional double bike_speed = 5;
    optional int32 bike_distance =6;
    optional double car_speed = 7;
    optional int32 car_distance =8;
}


message JourneysRequest {
    required string origin = 1;
    required string destination = 2;
    repeated string datetimes = 3;
    required bool clockwise = 4;
    repeated string forbidden_uris = 7;
    required int32 max_duration = 11;
    required int32 max_transfers = 12;
    optional StreetNetworkParams streetnetwork_params = 13;
}

message PlacesNearbyRequest {
    required string uri = 1;
    required double distance = 2;
    repeated NavitiaType types = 3;
    required int32 depth = 4;
}

message PTRefRequest {
    required NavitiaType requested_type = 1;
    required string filter = 2;
    required int32 depth = 3;
}

message Request {
    required API requested_api = 1;

    optional PlacesRequest places = 2;
    optional NextStopTimeRequest next_stop_times = 4;
    optional PlacesNearbyRequest places_nearby = 5;
    optional JourneysRequest journeys = 6;
    optional PTRefRequest ptref = 7;
}
